### ISSUE TITLE
(v2.4.8) Log output does not automatically create directory when it doesn't exist

### ISSUE THREAD

**Issue by OneOfTheirs** [doctest/doctest#676]
## Description

When running with `-r=junit -o=reports/log.xml`, if `reports` folder does not exist, no xml file will be actually generated. I am not sure what's behind this behavior, but gtest framework will create the folder if it doesn't exist.

### Steps to reproduce
just run the exe with `-r=junit -o=reports/log.xml`

### Extra information
* doctest version: **v2.4.8**
* Operating System: **ubuntu 18.04**
* Compiler+version: **gcc 7.5**

**Comment by philbucher** [comment]
this could be solved (I think relatively simple) with C++17 `std::filesystem`
afaik we still support C++11, which makes it a bit tedious as creating a folder is platform dependent.

Maybe this could be a feature enabled only when using C++17? 

BTW gcc supports `std::filesystem` only from version 8 :/

**Comment by OneOfTheirs** [comment]
I agree that it’s not very simply to automatically create the directory, but I find it very useful to be able to do so. Mainly because more and more people are doing out-of-source build now, when they do clean build, every sub folder is deleted. Sure, they just need to recreate the folder in their build system (that’s what I did in cmake) for every build regardless the existence of the directory.

**Comment by philbucher** [comment]
Ok I understand, makes sense to me

Can you please provide a minimal working example?

I think I should be able to solve this

**Comment by cdeln** [comment]
I suggest we close as a non-fix due to the following reasons

1) Easy workaround, just create the directory before running the tests
2) Implementation with C++ requires a recent standard. This implies that doctest will create or not create the output directory depending on which standard used when compiling the binary, which is frankly more confusing that saying that it does not create the output directory

**Comment by OneOfTheirs** [comment]
> I suggest we close as a non-fix due to the following reasons
> 
> 1. Easy workaround, just create the directory before running the tests
> 2. Implementation with C++ requires a recent standard. This implies that doctest will create or not create the output directory depending on which standard used when compiling the binary, which is frankly more confusing that saying that it does not create the output directory

I totally agree on not fixing it. However, silently doing something user never expects is not an acceptable behavior. Simply add a "WARNING, log output directory does not exist, failed to create logs" or something like that should be sufficient. In fact, when users specific to output log to a file, it should immediately fail when directly does not exist. It can be done in many ways.

if such warning exists, it would have saved me at least 2 hours trying to debug the problem deep into our CI.
